vendor/src/github.com/docker/swarmkit/api/ca.proto

```diff
@@ -43,6 +43,9 @@ message IssueNodeCertificateRequest {
 	// Token represents a user-provided string that is necessary for new
 	// nodes to join the cluster
 	string token = 3;
+
+	// Availability allows a user to control the current scheduling status of a node.
+	string availability = 4;
```

Comment:
This really isn't the right place for it.

Agree with the concept of providing a `NodeSpec` at join time.